test/functional/uri_options_spec_tests.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const fs = require('fs');
+const expect = require('chai').expect;
+const parse = require('mongodb-core').parseConnectionString;
+
+describe('URI Options (spec)', function() {
+  fs
+    .readdirSync(`${__dirname}/spec/uri-options`)
+    .filter(filename => filename.match(/\.json$/))
+    .forEach(filename => {
+      const specString = fs.readFileSync(`${__dirname}/spec/uri-options/${filename}`, 'utf8');
+      const specData = JSON.parse(specString);
+
+      describe(filename, () => {
+        specData.tests.forEach(test => {
+          const itFn = test.warning ? it.skip : it;
```

Comment:
do we plan on returning to implement the warning tests?

Reply:
Yes, I filed NODE-1881 for this.